### PR TITLE
Add agent dockerfile

### PR DESCRIPTION
**Pull request by gbstan92**
Added dockerfile for agent 
Added Azure Devops Pipeline that automatically publishes images when merged to master.

### CODE REVIEW

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Publish Docker Jenkins Images
+trigger:
+  branches:
```
Comment (naikajah) [inline azure-pipelines.yaml:3]: You can include -> `batch: true` after trigger like
```
trigger:
  batch: true
  branches:
    include:

```

Comment (gbstan92) [reply azure-pipelines.yaml:3]: done

Comment (gbstan92) [reply azure-pipelines.yaml:3]: done

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Publish Docker Jenkins Images
```
Comment (timja) [inline azure-pipelines.yaml:1]: We use acr tasks for this, you don’t need a pipeline

Comment (gbstan92) [reply azure-pipelines.yaml]: I've created the 2 acr tasks. here's the output for one of them:

bogdans$ az acr task create --registry $ACR_NAME --name jenkins-agent --image jenkins/agent:{{.Run.ID}} --context https://github.com/hmcts/cnp-jenkins-docker.git --file master\Dockerfile --git-access-token $GIT_PAT --subscription DCD-CNP-PROD
{- Finished ..
  "agentConfiguration": {
    "cpu": 2
  },
  "agentPoolName": null,
  "creationDate": "2020-06-18T11:03:57.818028+00:00",
  "credentials": null,
  "id": "/subscriptions/8999dec3-0104-4a27-94ee-6588559729d1/resourceGroups/rpe-acr-prod-rg/providers/Microsoft.ContainerRegistry/registries/hmctspublic/tasks/jenkins-agent",
  "identity": null,
  "location": "uksouth",
  "name": "jenkins-agent",
  "platform": {
    "architecture": "amd64",
    "os": "linux",
    "variant": null
  },
  "provisioningState": "Succeeded",
  "resourceGroup": "rpe-acr-prod-rg",
  "status": "Enabled",
  "step": {
    "arguments": [],
    "baseImageDependencies": null,
    "contextAccessToken": null,
    "contextPath": "https://github.com/hmcts/cnp-jenkins-docker.git",
    "dockerFilePath": "masterDockerfile",
    "imageNames": [
      "jenkins/agent:{{.Run.ID}}"
    ],
    "isPushEnabled": true,
    "noCache": false,
    "target": null,
    "type": "Docker"
  },
  "tags": null,
  "timeout": 3600,
  "trigger": {
    "baseImageTrigger": {
      "baseImageTriggerType": "Runtime",
      "name": "defaultBaseimageTriggerName",
      "status": "Enabled",
      "updateTriggerEndpoint": null,
      "updateTriggerPayloadType": "Default"
    },
    "sourceTriggers": [
      {
        "name": "defaultSourceTriggerName",
        "sourceRepository": {
          "branch": "master",
          "repositoryUrl": "https://github.com/hmcts/cnp-jenkins-docker.git",
          "sourceControlAuthProperties": null,
          "sourceControlType": "Github"
        },
        "sourceTriggerEvents": [
          "commit"
        ],
        "status": "Enabled"
      }
    ],
    "timerTriggers": null
  },
  "type": "Microsoft.ContainerRegistry/registries/tasks"
}

File: agent/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+FROM jenkins/inbound-agent:${version}
+
+ARG user=jenkins
+
+USER root
+
+    #Install Prerequisites
+RUN apt-get update && \
+	apt-get -y install \
```
Comment (timja) [inline agent/Dockerfile:11]: can you fix the mixed tabs and spaces in this file, might be best to add an editorconfig file to the repo

Comment (gbstan92) [reply agent/Dockerfile]: Fixed fmt issue and added editorconfig file

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# EditorConfig is awesome: https://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# tab indentation
+[Dockerfile]
+indent_style = tab
```
Comment (timja) [inline .editorconfig:8]: Don’t you want spaces?

File: agent/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    chmod +x /usr/bin/chromedriver && \
+    curl https://selenium-release.storage.googleapis.com/3.13/selenium-server-standalone-3.13.0.jar --output selenium-server-standalone.jar && \
+    curl http://www.java2s.com/Code/JarDownload/testng/testng-6.8.7.jar.zip --output testng-6.8.7.jar.zip && \
+    unzip testng-6.8.7.jar.zip && \
+    rm \
+    testng-6.8.7.jar.zip \
+    chromedriver_linux64.zip && \
+    apt-get -y remove --purge \
+		unzip \
+		&& \
```
Comment (timja) [inline agent/Dockerfile:44]: still more tabs, you might want to reformat this file

Comment (gbstan92) [reply agent/Dockerfile:44]: done

File: .editorconfig (hunk 4 above)
Comment (gbstan92) [reply .editorconfig]: done

File: agent/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    chromedriver_linux64.zip && \
+    apt-get -y remove --purge unzip && \
+    apt-get -y autoremove
+
+USER ${user}
+
+   #Install tfenv
+RUN	git clone https://github.com/tfutils/tfenv.git ~/.tfenv && \
+    echo 'PATH=${HOME}/.tfenv/bin:${PATH}' >> ~/.bashrc && \
+	. ~/.bashrc
```
Comment (timja) [inline agent/Dockerfile:50]: more tabs

File: agent/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    rm \
+    testng-6.8.7.jar.zip \
+    chromedriver_linux64.zip && \
+    apt-get -y remove --purge unzip && \
+    apt-get -y autoremove
+
+USER ${user}
+
+   #Install tfenv
+RUN	git clone https://github.com/tfutils/tfenv.git ~/.tfenv && \
```
Comment (timja) [inline agent/Dockerfile:48]: more tabs,could you check the file in github ui / command line please, i assumed you would format the file, and I wouldn't need to comment on every line =/

File: agent/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,52 @@
+ARG version=4.3-4-jdk11
+
+FROM jenkins/inbound-agent:${version}
+
+ARG user=jenkins
+
+USER root
+
+    #Install Prerequisites
```
Comment (timja) [inline agent/Dockerfile:9]:
```suggestion
# Install Prerequisites
```

File: agent/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+RUN apt-get update && \
+    apt-get -y install \
+        curl \
+        git \
+    #Install Node
```
Comment (timja) [inline agent/Dockerfile:14]:
```suggestion
    # Install Node
```

File: agent/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        nodejs \
+        yarn \
+        && \
+    #Install Azure CLI
```
Comment (timja) [inline agent/Dockerfile:18]:
```suggestion
    # Install Azure CLI
```

File: agent/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    curl -sL https://aka.ms/InstallAzureCLIDeb | bash && \
+    #Install Headless Chrome
```
Comment (timja) [inline agent/Dockerfile:20]:
```suggestion
    # Install Headless Chrome
```

File: agent/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    unzip testng-6.8.7.jar.zip && \
+    rm \
+    testng-6.8.7.jar.zip \
+    chromedriver_linux64.zip && \
+    apt-get -y remove --purge unzip && \
+    apt-get -y autoremove
+
+USER ${user}
+
+   #Install tfenv
```
Comment (timja) [inline agent/Dockerfile:47]:
```suggestion
# Install tfenv
```

File: agent/Dockerfile [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+RUN	git clone https://github.com/tfutils/tfenv.git ~/.tfenv && \
+    echo 'PATH=${HOME}/.tfenv/bin:${PATH}' >> ~/.bashrc && \
+	. ~/.bashrc
+
+ENTRYPOINT ["/bin/bash"]
```
Comment (timja) [inline agent/Dockerfile:52]: I don't think we want this?

File: agent/Dockerfile (hunk 6 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 7 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 8 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 9 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 11 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 12 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

File: agent/Dockerfile (hunk 13 above)
Comment (gbstan92) [reply agent/Dockerfile]: done

**Pull request merged**